python/src/spark_rapids_ml/knn.py

```diff
@@ -686,3 +695,390 @@ def read(cls) -> MLReader:
         raise NotImplementedError(
             "NearestNeighborsModel does not support loading/loading, just re-fit the estimator to re-create a model."
         )
+
+
+class ApproximateNearestNeighborsClass(_CumlClass):
+
+    @classmethod
+    def _param_mapping(cls) -> Dict[str, Optional[str]]:
+        return {
+            "k": "n_neighbors",
+            "algorithm": "algorithm",
+            "algo_params": "algo_params",
+        }
+
+    def _get_cuml_params_default(self) -> Dict[str, Any]:
+        return {"n_neighbors": 5, "verbose": False, "algorithm": "ivfflat"}
+
+    def _pyspark_class(self) -> Optional[ABCMeta]:
+        return None
+
+
+class DictTypeConverters(TypeConverters):
+    @staticmethod
+    def _toDict(value: Any) -> Dict[str, Any]:
+        """
+        Convert a value to a Dict type for Param typeConverter, if possible.
+        """
+        if isinstance(value, Dict):
+            return {TypeConverters.toString(k): v for k, v in value.items()}
+        raise TypeError("Could not convert %s to Dict[str, Any]" % value)
+
+
+class _ApproximateNearestNeighborsParams(_NearestNeighborsCumlParams):
+    def __init__(self) -> None:
+        super().__init__()
+        self._setDefault(algorithm="ivfflat")
+        self._setDefault(algo_params=None)
+
+    algorithm = Param(
+        Params._dummy(),
+        "algorithm",
+        "The algorithm to use for approximate nearest neighbors search.",
+        typeConverter=TypeConverters.toString,
+    )
+
+    algo_params = Param(
+        Params._dummy(),
+        "algo_params",
+        "The parameters to use to set up a neighbor algorithm.",
+        typeConverter=DictTypeConverters._toDict,
+    )
+
+
+class ApproximateNearestNeighbors(
+    ApproximateNearestNeighborsClass, _CumlEstimator, _ApproximateNearestNeighborsParams
+):
+    """
+    IVF_FLAT retrieves the k approximate nearest neighbors in item vectors for each query
```

Comment:
Shouldn't we have doc string similar to other algos?   With explanation of params, example, etc.   For api docs.

Reply:
Added the docstring.